src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState, ...action.extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        currentState = Object.keys(currentState)
+          .filter(key => !action.keysToRemove.includes(key))
+          .reduce((newState, key) => {
+            newState[key] = currentState[key];
+
+            return newState;
+          }, {});
```

Comment:
The `removeProperties` case can be simplified by using `Object.fromEntries` along with `Object.entries` and `filter`. This would make the code more concise and easier to read. Consider using: `currentState = Object.fromEntries(Object.entries(currentState).filter(([key]) => !action.keysToRemove.includes(key)));`